internal/cmd/events.go

```diff
@@ -79,14 +83,18 @@ func retrieveClusterEvents(kubeConfig string) error {
 		return &ErrorWithMsg{"failed to get a list of namespaces", err}
 	}
 
+	if namespaceFilter != "" {
+		namespaceFilter = "metadata.namespace==" + namespaceFilter
+	}
+
 	notes := make(chan note)
 
 	// Start one Go routine per namespace to watch for events
 	for i := range namespaces {
```

Comment:
I am wondering whether it would be a more resource friendly implementation to only start the Go routine for the namespace that is actually requested by the user. Or in other words, I think it would be better when inside the for loop to check the namespace name against the given namespace name and to skip (`continue`) all other namespaces that do not match.